File: 프로그래머스/2/42747. H－Index/H－Index.py
def solution(citations: list) -> int:
    answer = 0
    citations.sort()
    citations.reverse()
    
    for i in range(0, len(citations)):
        answer += 1
        if (citations[i] < answer):
            return answer - 1
        elif (citations[i] == answer):
            return answer
        
    return answer

File: 프로그래머스/2/42747. H－Index/H－Index.java
import java.util.Arrays;

public class Solution {
    public int solution(int[] citations) {
        int answer = 0;
        Arrays.sort(citations);
        for (int i = citations.length - 1; i >= 0; i--) {
            answer++;
            if (citations[i] < answer) {
                return answer -= 1;
            } else if (citations[i] == answer) {
                return answer;
            }
        }
        return answer;
    }
}
